pkg/cmd/admin/node/node.go

```diff
@@ -0,0 +1,399 @@
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/util"
+
+	"github.com/openshift/origin/pkg/cmd/util/clientcmd"
+)
+
+const ManageNodeCommandName = "manage-node"
+
+type NodeConfig struct {
+	DefaultNamespace string
+	Kclient          *client.Client
+	Writer           io.Writer
+
+	Mapper            meta.RESTMapper
+	Typer             runtime.ObjectTyper
+	RESTClientFactory func(mapping *meta.RESTMapping) (resource.RESTClient, error)
+	Printer           func(mapping *meta.RESTMapping, noHeaders bool) (kubectl.ResourcePrinter, error)
+
+	Operations []string
+	NodeNames  []string
+
+	// Supported operations
+	Schedulable bool
+	Evacuate    bool
+	ListPods    bool
+
+	// Optional arguments
+	Selector    util.StringFlag
+	PodSelector util.StringFlag
+	DryRun      bool
+}
+
+const manageNodeLongDesc = `Manage node operations: schedulable, evacuate and list-pods.
+
+Examples:
+
+	# Block accepting any new pods on nodes
+	$ %[1]s <mynode> --schedulable=false
+
+	# Move selected pods to a different node
+	$ %[1]s <mynode> --evacuate --pod-selector="<service=myapp>"
+
+	# List all pods on nodes
+	$ %[1]s <mynode1> <mynode2> --list-pods
+
+	# Mark selected nodes as schedulable
+	$ %[1]s --selector="<env=dev>" --schedulable=true
+`
+
+func NewCommandManageNode(f *clientcmd.Factory, commandName, fullName string, out io.Writer) *cobra.Command {
+	cfg := &NodeConfig{}
+
+	cmd := &cobra.Command{
+		Use:   commandName,
+		Short: "Manage node operations: schedulable, evacuate, list-pods",
+		Long:  fmt.Sprintf(manageNodeLongDesc, fullName),
+		Run: func(c *cobra.Command, args []string) {
+
+			defaultNamespace, err := f.DefaultNamespace()
+			kcmdutil.CheckErr(err)
+
+			_, kc, err := f.Clients()
+			kcmdutil.CheckErr(err)
+
+			mapper, typer := f.Object()
+
+			cfg.DefaultNamespace = defaultNamespace
+			cfg.Kclient = kc
+			cfg.Writer = out
+			cfg.Mapper = mapper
+			cfg.Typer = typer
+			cfg.RESTClientFactory = f.Factory.RESTClient
+			cfg.Printer = f.Printer
+			cfg.Operations = []string{}
+			cfg.NodeNames = []string{}
+
+			if c.Flag("schedulable").Changed {
+				cfg.Operations = append(cfg.Operations, "schedulable")
+			}
+			if c.Flag("evacuate").Changed {
+				cfg.Operations = append(cfg.Operations, "evacuate")
+			}
+			if c.Flag("list-pods").Changed {
+				cfg.Operations = append(cfg.Operations, "list-pods")
+			}
+
+			if len(args) != 0 {
+				cfg.NodeNames = append(cfg.NodeNames, args...)
+			}
+
+			errList := []error{}
+			if errList = cfg.ValidateParams(); len(errList) != 0 {
+				err := fmt.Errorf("%v", errList)
+				kcmdutil.CheckErr(kcmdutil.UsageError(c, err.Error()))
+			}
+
+			err = cfg.RunManageNode()
+			kcmdutil.CheckErr(err)
+		},
+	}
+	flags := cmd.Flags()
+
+	flags.BoolVar(&cfg.Schedulable, "schedulable", false, "Control pod schedulability on the node.")
+	flags.BoolVar(&cfg.Evacuate, "evacuate", false, "Migrate all/selected pods on the node.")
+	flags.BoolVar(&cfg.ListPods, "list-pods", false, "List all/selected pods on the node.")
```

Comment:
If I don't, I need them.  Otherwise, listing pods here is decidedly less powerful.